bessctl/conf/testing/run_module_tests.bess

```diff
@@ -0,0 +1,166 @@
+import scapy.all as scapy
+import sys
+import os
+import glob
+import socket
+import time
+import sugar
+from time import gmtime, strftime
+
+## HANDY UTILS AND GLOBAL VARS ##
+SOCKET_PATH = '/tmp/bess_unix_'
+SCRIPT_STARTTIME = strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+
+# Generate a UnixSocketPort and a Socket to talk to it
+
+
+def gen_socket_and_port(sockname):
+    socket_port = UnixSocketPort(
+        name=sockname,
+        path='@' +
+        SOCKET_PATH +
+        sockname)
+
+    s = socket.socket(socket.AF_UNIX, socket.SOCK_SEQPACKET)
+    s.settimeout(5)  # five second timeout
+    s.connect('\0' + SOCKET_PATH + sockname)
+    return socket_port, s
+
+# Craft a packet with the specified IP addresses
+# All the other fields -- Ether, ports -- are dummy values.
+
+
+def gen_packet(proto, src_ip, dst_ip):
+    eth = scapy.Ether(src='02:1e:67:9f:4d:ae', dst='06:16:3e:1b:72:32')
+    ip = scapy.IP(src=src_ip, dst=dst_ip)
+    udp = proto(sport=10001, dport=10002)
+    payload = 'helloworld'
+    pkt = eth / ip / udp / payload
+    return str(pkt)
+
+
+def pkt_str(pkt):
+    if(pkt is None):
+        return "None"
+    else:
+        return pkt.summary()
+
+# Convert an IP address to an integer
+
+
+def aton(ip):
+    return socket.inet_aton(ip)
+
+# These are just for the crash test
+crash_test_packets = [
+    gen_packet(scapy.UDP, '172.16.100.1', '10.0.0.1'),
+    gen_packet(scapy.UDP, '172.12.55.99', '12.34.56.78'),
+    gen_packet(scapy.UDP, '172.12.55.99', '10.0.0.1'),
+    gen_packet(scapy.UDP, '172.16.100.1', '12.34.56.78'),
+    gen_packet(scapy.TCP, '172.12.55.99', '12.34.56.78'),
+    gen_packet(scapy.UDP, '192.168.1.123', '12.34.56.78')
+]
+
+## TEST FUNCTIONS ##
+
+# All this does is subject a module to load and make sure it
+# doesn't crash
+
+
+def crash_test(module, num_input_ports, num_output_ports):
+    for i in range(num_input_ports):
+        src = Source()
+        src -> Rewrite(templates=crash_test_packets) -> i:module
+    for i in range(num_output_ports):
+        sink = Sink()
+        module:i -> sink
+
+    time.sleep(5)
+
+
+## RUN TEST LOOP ##
+
+path = os.path.dirname(os.path.realpath(sys.argv[0]))
+for file in glob.glob(path + "/conf/testing/module_tests/*.py"):
+    bess.reset_all()
+    CRASH_TEST_INPUTS = []
+    OUTPUT_TEST_INPUTS = []
+    CUSTOM_TEST_FUNCTIONS = []
+
+    print("-- Running Module Tests for " + file + " --")
+    while True:  # wrap in this scoping to make sure variables don't leak into this script
```

Comment:
http://stackoverflow.com/questions/291978/short-description-of-scoping-rules
This suggests I should wrap it in a def loadtest() -- I'll try that and see if it works.